### PR TITLE
Remove redundant OperationKinds

### PR DESCRIPTION
**Pull request by mavasani**
FieldInitializerInCreation and PropertyInitializerInCreation are no longer used, so this change removes them.
Fixes https://github.com/dotnet/roslyn/issues/19300

### CODE REVIEW

**Comment by mavasani** [comment]
This change depends on https://github.com/dotnet/roslyn/pull/20689. Tests are expected to fail till that PR is merged.

File: src/Compilers/Core/Portable/Operations/OperationKind.cs [hunk 1]
```diff
@@ -157,13 +157,12 @@ public enum OperationKind
 
         // Operations that are constituents of statements, expressions, or declarations.
 
-        /// <summary>Indicates an <see cref="IFieldInitializer"/>.</summary>
-        FieldInitializerInCreation = 0x400,
+
+        // Unused 0x400 and 0x402
+
         /// <summary>Indicates an <see cref="IFieldInitializer"/>.</summary>
         FieldInitializerAtDeclaration = 0x401,
```
Comment (heejaechang) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: should we change the name as well then? like removing AtDeclaration

Comment (mavasani) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: Sounds like a good idea. Any objections from others? Otherwise, I will make this change.

**Comment by heejaechang** [comment]
👍

**Review by CyrusNajmabadi — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM

File: src/Compilers/Core/Portable/Operations/OperationKind.cs [hunk 2]
```diff
@@ -157,16 +157,15 @@ public enum OperationKind
 
         // Operations that are constituents of statements, expressions, or declarations.
 
+
+        // Unused 0x400 and 0x402
```
Comment (333fred) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: Would prefer if this said something about those numbers being available for use.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/Core/Portable/Operations/OperationKind.cs (hunk 2 above)
Comment (mavasani) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: Sure, will update in the next PR.

**Comment by cston** [comment]
LGTM

**Review by jinujoseph — APPROVED** [review]

**Comment by mavasani** [comment]
windows_debug_unit32_prtest failed due to flaky test https://github.com/dotnet/roslyn/issues/20106

https://ci.dot.net/job/dotnet_roslyn/job/features_ioperation/job/windows_debug_unit32_prtest/227/

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_unit32_prtest please

**Pull request merged**